File: android/src/main/kotlin/com/rudo/hybrid_stripe_payments/channel/methods/WalletStripeMethodType.kt
package com.rudo.hybrid_stripe_payments.channel.methods

import com.rudo.hybrid_stripe_payments.core.stripe.StripePaymentSheetType
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

enum class WalletStripeMethodType(val rawValue: String) {
    Initialize("initialize"),
    ShowPaymentSheet("showPaymentSheet"),
    ShowManageCreditCardsSheet("setupStripeSheet");

    fun handle(
        call: MethodCall,
        result: MethodChannel.Result
    ) {
        when (this) {
            Initialize -> WalletStripeInitHandler().handle(call, result)
            ShowPaymentSheet -> WalletStripeShowSheetHandler().handle(call, result, StripePaymentSheetType.PAYMENT)
            ShowManageCreditCardsSheet -> WalletStripeShowSheetHandler().handle(call, result, StripePaymentSheetType.SETUP)
        }
    }

    companion object {
        fun fromRawValue(method: String?): WalletStripeMethodType? {
            return entries.firstOrNull { it.rawValue == method }
        }
    }
}

File: ios/Classes/Channel/Methods/WalletStripeMethodType.swift
//
//  WalletStripeMethodType.swift
//
//
//  Created by Miguel Ángel Soto González on 20/10/25.
//

import Flutter

enum WalletStripeMethodType: String {
    case initialize
    case showPaymentSheet
    case showManageCreditCardsSheet = "setupStripeSheet"

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch self {
        case .initialize:
            return WalletStripeInitHandler().handle(call, result: result)
        case .showPaymentSheet:
            return WalletStripeShowPaymentSheetHandler().handle(call, result: result)
        case .showManageCreditCardsSheet:
            return WalletStripeShowManageCreditCardSheetHandler().handle(call, result: result)
        }
    }
}
